### PR TITLE
`Communication`: Add private channel icon to sidebar

### LINKED ISSUE #9149: `Communication`: Missing private channel icon in sidebar
### Is your feature request related to a problem?

Although the private channel icon (lock icon) is visible in both the conversation view and the create channel dialog, it is not directly visible in the sidebar. Users cannot see whether the channel is private without clicking on it.

### Describe the solution you'd like

The private channel icon should be added next to each private channel name in the sidebar view, appearing before the channel title.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by asliayk**
#### General

- [X] This is a small issue that I tested locally.
- [X] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [X] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [X] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Currently, users cannot easily distinguish between private and public channels in the sidebar without clicking on each channel, since there is no private channel icon.
Closes #9149 

### Description

Added a lock icon next to each private channel name in the sidebar, positioned before the channel title.

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Navigate to the Communication section of a course.
2. In the general section of the sidebar, check if there is any private channel. If not, create a private channel by clicking the "+" button.
3. Verify that the private channel has a lock icon.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2

### Screenshots

**OLD**
image

**NEW**
image

## Summary by CodeRabbit

- **New Features**
	- Introduced a new icon for private conversations, enhancing visual differentiation between public and private conversations.
	- Improved the logic for assigning icons based on conversation visibility, providing clearer user feedback.
- **Bug Fixes**
	- Resolved inconsistencies in icon representation for different conversation types.

### CODE REVIEW

**Review by edkaya — DISMISSED** [review]
code

File: src/main/webapp/app/overview/course-overview.service.ts [hunk 1]
```diff
@@ -313,7 +314,7 @@ export class CourseOverviewService {
             title: this.conversationService.getConversationName(conversation) ?? '',
             id: conversation.id ?? '',
             type: conversation.type,
-            icon: getAsChannelDTO(conversation)?.name === 'announcement' ? this.faBullhorn : this.faHashtag,
+            icon: getAsChannelDTO(conversation)?.isPublic === false ? this.faLock : getAsChannelDTO(conversation)?.name === 'announcement' ? this.faBullhorn : this.faHashtag,
```
Comment (Strohgelaender) [inline src/main/webapp/app/overview/course-overview.service.ts:317]: This is now quite long, can you move this into a method?

**Review by Strohgelaender — DISMISSED** [review]

**Review by JohannesStoehr — DISMISSED** [review]
Code

**Review by egekurt123 — APPROVED** [review]
Code LGTM

File: src/main/webapp/app/overview/course-overview.service.ts [hunk 2]
```diff
@@ -75,6 +76,7 @@ export class CourseOverviewService {
 
     faBullhorn = faBullhorn;
     faHashtag = faHashtag;
+    faLock = faLock;
```
Comment (MaximilianAnzinger) [inline src/main/webapp/app/overview/course-overview.service.ts:79]: We could improve the data encapsulation here. At least making it read only would already be a first step.

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]

**Review by Strohgelaender — APPROVED** [review]

**Review by SimonEntholzer — APPROVED** [review]
code

**Review by sarpsahinalp — APPROVED** [review]
Code lgtm

**Review by pzdr7 — APPROVED** [review]
LGTM on TS2 👍

**Review by krusche — APPROVED** [review]

**Pull request merged**